src/librustc/ty/mod.rs

```diff
@@ -2749,7 +2749,12 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 fn describe_def<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                           def_id: DefId)
                           -> Option<Def> {
-    ty::queries::describe_def::get(tcx, DUMMY_SP, def_id)
+   let id = tcx.hir.as_local_node_id(def_id).unwrap();
+   let item = tcx.hir.expect_item(id);
+   match item.node{
+       hir::ItemTrait(..) => Some(Def::Trait(def_id)),
```

Comment:
This appears to be correct, although `expect_item` above is too restrictive and there's a dozen or so cases left to handle, of course.

However, I'd like to merge the original PR, without this change, as we're *not using this right now*.